src/main/java/duke/CommandParser.java

```diff
@@ -0,0 +1,150 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.action.Action;
+import duke.action.Delete;
+import duke.action.Echo;
+import duke.action.Farewell;
+import duke.action.Mark;
+import duke.action.Match;
+import duke.action.MyList;
+import duke.action.TaskList;
+import duke.action.Unmark;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.NoIndexException;
+import duke.exception.UnknownCommandException;
+import duke.exception.WrongDateFormatException;
+import duke.exception.WrongDateOrderingException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Parses user commands and performs corresponding actions in the Duke application.
+ */
+
+public class CommandParser {
+
+    /**
+     * Parses user commands and performs corresponding actions in the Duke application.
+     */
+    public static Action parseCommand(String command, TaskList taskList) throws
+            EmptyDescriptionException, NoIndexException, UnknownCommandException,
+            WrongDateFormatException, WrongDateOrderingException {
```

Comment:
Since these are all subclasses of `DukeException` why not use `throws DukeException` instead?